### ISSUE TITLE
Synthetic Dataset

### ISSUE THREAD

**Issue by Olivier-tl** [Olivier-tl/MO-DAGAN#8]
-Create a pytorch dataset that returns synthetic examples from a trained GAN model

**Comment by wwwnev** [comment]
maybe this subtask could be interesting too :
- Handling 2 datasets (original + synthetic) for classification training.

We could concatenate the dataloaders : https://discuss.pytorch.org/t/train-simultaneously-on-two-datasets/649
or
We could chain IterableDatasets. https://pytorch.org/docs/master/data.html#torch.utils.data.ChainDataset

Any preferences, better solutions ? Maybe this should have it's own seperate issue :P

**Comment by Olivier-tl** [comment]
I would use [ConcatDataset](https://pytorch.org/docs/master/data.html#torch.utils.data.ConcatDataset) more than ChainDataset, because i think our dataset are not iterable but they are indexable!

Also, i'm wondering if/how a dataloader is able to get iid/shuffled data from an iterable dataset... 🤔

And of course you could open a new issue just for that too 🙂
